### PR TITLE
Complete documentation rewrite

### PR DESCRIPTION
**Pull request by Luc45**
**PS: Greg, please, don't review this on Saturday, nor Sunday.**

Definetly not done by O1 Pro. Trust me bro.

### What Changed
- Complete reorganization of the documentation structure
- New sidebar navigation
- Consolidated duplicate content (CLI/Dashboard stuff was everywhere)
- Added missing sections for advanced usage and test execution
- Made everything more consistent and interconnected

### Testing Instructions
- Run this branch locally
- See the difference in the organization in the sidebar, see if you like it
- Navigate the documentation
- Compare this proposed documentation with the old documentation, preferably PAGE BY PAGE, as there are some pages in the old documentation with graphics and things like that that were not ported over.. Ideally, we should try to make this new documentation a bit more visual in some parts, with some graphs, etc

### Known TODOs
- Port over graphics from old docs
- Add more visuals where it makes sense (especially for complex concepts)
- Double-check all cross-links work
- Make sure we didn't lose any crucial info in the reorganization

O1 is good but I had to stick with it for over 5h until 2am because I didn't want to lose the huge context I had while generating this doc. I generated this by first figuring out a structure that made sense with it, then generating it page by page, then today, in the morning, coming up with all kinds of crazy scripts to fix the internal links. O1 is good, but I just don't want to make it feel like it's magic. There was a looot of work.

### CODE REVIEW

File: docs/managed-tests/plugin-check.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Plugin Check Tests
```
Comment (singerb) [inline docs/managed-tests/plugin-check.md]: I'll be the AI skeptic and say I don't love this one; it's both longer and more verbose than the original, while not containing all the same info (which specific checks do we run, and the link to the review guidelines).

File: docs/managed-tests/plugin-check.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Plugin Check tests run the [wordpress.org automated plugin check](#) tool against your extension. This tool helps identify issues and guidelines violations aligned with the WordPress.org plugin directory standards. While this test is optional and not automatically triggered by QIT for WooCommerce Marketplace submissions, it can be valuable if you also distribute your extension on wordpress.org or simply want to align with broader WordPress best practices.
```
Comment (singerb) [inline docs/managed-tests/plugin-check.md]: Missing link.

File: docs/managed-tests/security.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Security Tests
```
Comment (singerb) [inline docs/managed-tests/security.md]: Same comment on this one as the plugin check; I think the original was better.

File: docs/managed-tests/validation.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+- **Plugin Headers:** Ensures that important headers such as `Requires PHP`, `Requires at least`, `Tested up to`, `WC requires at least`, and `WC tested up to` are present and valid. Missing or incorrect headers can lead to confusion or compatibility issues.
+- **WooCommerce Feature Support:** Verifies any declared incompatibilities with WooCommerce features like High Performance Order Storage (HPOS) or Cart and Checkout blocks. Explicit incompatibilities are flagged, encouraging you to support these features where possible.
+- **Theme Templates (For Themes):** Checks if your theme has outdated WooCommerce templates. If found, this results in a warning, guiding you to update templates for the latest WooCommerce standards.
+
+## Possible Outcomes
+
+- **Success:** All required metadata is present and correct, and no invalid headers or incompatible feature declarations were found.
+- **Warning:** Some headers may be missing or incorrect, or you declared incompatibility with features that you should ideally support.
+- **Failed (for Themes):** Outdated templates or critical metadata issues prevent passing the validation.
```
Comment (singerb) [inline docs/managed-tests/validation.md]: This is inaccurate, it's a warning right now.

File: docs/managed-tests/validation.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Validation Tests
```
Comment (singerb) [inline docs/managed-tests/validation.md]: Again, missing links here. This rewrite is arguably ok, as the original is perhaps a bit too verbose, but I'm still iffy.

**Comment by singerb** [comment]
Oh, one overall note: I thought we wanted things `In sentence case` rather than `In Title Case`, but PCYsg-5n-p2me that title-case is fine. This means the PR is probably fine, but if you can feed the style guide to the AI too maybe it can follow it automatically?

File: docs/core-concepts/test-types-overview.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# Understanding Test Types in QIT
+
+QIT supports multiple kinds of tests designed to ensure the quality, compatibility, and security of WooCommerce extensions. These tests fall into two main categories: **Managed Tests** and **Custom E2E Tests**.
+
+Why does this matter?  
+By familiarizing yourself with these test types, you can leverage QIT to catch issues early, maintain compatibility with evolving WordPress and WooCommerce versions, and deliver a reliable experience to merchants and their customers.
```
Comment (zhongruige) [inline docs/core-concepts/test-types-overview.md]: I would recommend re-working this a bit, and drop the `Why does this matter?` heading, so it's something like this instead:

```suggestion
By utilizing test types provided by QIT, you can leverage QIT to catch issues early, maintain compatibility with evolving WordPress and WooCommerce versions, and deliver a reliable experience to merchants and their customers.
```

File: docs/core-concepts/test-types-overview.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+## Managed Tests
+
+Definition:
```
Comment (zhongruige) [inline docs/core-concepts/test-types-overview.md]: I would drop the "Definition" header here and below, it feels redundant.

File: docs/managed-tests/plugin-check.md (hunk 2 above)
Comment (Luc45) [reply docs/managed-tests/plugin-check.md]: Fixed

**Comment by Luc45** [comment]
> Oh, one overall note: I thought we wanted things `In sentence case` rather than `In Title Case`, but PCYsg-5n-p2me that title-case is fine. This means the PR is probably fine, but if you can feed the style guide to the AI too maybe it can follow it automatically?

Fixed in https://github.com/woocommerce/qit-documentation/pull/50/commits/802a27ef5d976ad88cea0650460ba2b73bc241fe and https://github.com/woocommerce/qit-documentation/pull/50/commits/307844f61c3878b13a51134c17332fae01e40217

File: docs/managed-tests/plugin-check.md (hunk 1 above)
Comment (Luc45) [reply docs/managed-tests/plugin-check.md]: I've incorporated all the info from the old one, while keeping the "writing style" consistent with the rest of the documentation - which is one of the good things about this rewrite, to consolidate the style of writing. I'm open to any suggestions you might have, and feel free to own this and change it however you see fit, it's completely fine. Let me know if you like this version or not:

image

File: docs/managed-tests/security.md (hunk 3 above)
Comment (Luc45) [reply docs/managed-tests/security.md]: Agree 100%, that was an oversight. Let me know what you think of this version, and similar to the Plugin Check, feel free to modify it in any way shape or form.

image

File: docs/managed-tests/validation.md (hunk 5 above)
Comment (Luc45) [reply docs/managed-tests/validation.md]: Let me know what you think of this one (Updated screenshot just now, 14h30 UTC):
image

**Comment by Luc45** [comment]
I've also removed all "Next Steps" sections at the bottom of all pages. These links were disrupting the flow of the documentation and often linking to places that didn't make much sense. Now we let "Next" and "Previous" component at the bottom be the "guide":

image

I've only left a "Next Steps" on the "Installing" page, where it actually makes sense, since the next steps it points to authenticating and running the first managed test.

One thing that I still want to do (but doesn't need to a blocker to merging) is adding CLI options examples to some of the commands, especially around the managed tests, for instance, on the PHPCompatibility tests, we could mention `--min_php_version` and `--max_php_version`, etc.

**Comment by Luc45** [comment]
> but if you can feed the style guide to the AI too maybe it can follow it automatically?

`<irony>`Hey! That's not how AI works! Do you think things are easy like that...?!`</irony>`

Sometimes we just need to get a little creative, here's the series of prompts:

```
Write a PHP script that will recursively iterate over all files ending in ".md" in the current directory, and replace all headings to use "In sentence case" instead of "In Sentence Case".
 
Example:
 
# Security Tests
 
Becomes # Security Tests
 
## What Tools Are Used? becomes ## What tools are used?
 
etc
```

It gave a script like this:

```php
<?php

/**
 * Convert a heading string to sentence case.
 * Example: "What Tools Are Used?" -> "What tools are used?"
 *          "Security Tests"       -> "Security tests"
 */
function toSentenceCase($string) {
    $string = trim($string);
    // Convert entire string to lowercase, then uppercase first character
    $string = strtolower($string);
    $string = ucfirst($string);
    return $string;
}

/**
 * Process a single markdown file, converting headings to sentence case.
 */
function processFile($filename) {
    $content = file_get_contents($filename);
    $lines = explode("\n", $content);

    foreach ($lines as &$line) {
        // Match headings: One or more '#' followed by whitespace, then heading text
        if (preg_match('/^(#+\s+)(.*)$/', $line, $matches)) {
            $prefix = $matches[1]; // The '#' characters plus the trailing space
            $headingText = $matches[2];

            // Convert the heading text to sentence case
            $headingText = toSentenceCase($headingText);

            // Reassemble the line
            $line = $prefix . $headingText;
        }
    }

    // Save updated content back to the file
    file_put_contents($filename, implode("\n", $lines));
}

// Use recursive directory iteration to find all .md files
$directory = new RecursiveDirectoryIterator('.');
$iterator = new RecursiveIteratorIterator($directory);

foreach ($iterator as $file) {
    if ($file->isFile() && preg_match('/\.md$/i', $file->getFilename())) {
        processFile($file->getPathname());
    }
}

echo "Done processing markdown files.\n";
```

I've run the script and generated a git patch with the diff. Then I fed it the diff of the changes that the script caused, which was around 140kb of text, which is a lot, and asked it to analyse carefully all the changes to see if it's accurate. It gave me this:

```
While converting headings to sentence case, some brand names, acronyms, and terms that should remain capitalized have been inadvertently changed to lowercase. These include:

    Brand and Product Names:
        QIT: This acronym stands for "Quality Insights Toolkit" and is a product name. It should remain consistently capitalized as "QIT," not "Qit."
        WooCommerce: This brand name should remain "WooCommerce," not "woocommerce."
        WordPress: Always capitalized as "WordPress," not "wordpress."
        GitHub: Should remain "GitHub," not "github."
        WooCommerce Marketplace: Should stay "WooCommerce Marketplace," not "woocommerce marketplace."
        Cloudflare: When mentioned, "Cloudflare" should remain properly capitalized, not "cloudflare."
        AWS S3, GCP, or Azure Storage: These are brand/service names and acronyms. "AWS," "GCP," and "Azure Storage" should remain capitalized, not "aws s3," "gcp," or "azure storage."

    Acronyms and Technical Terms:
    These are commonly recognized acronyms and should remain in uppercase:
        CLI (Command Line Interface): Should remain "CLI," not "cli."
        UI (User Interface): Remain "UI," not "ui."
        API (Application Programming Interface): Remain "API," not "api."
        E2E (End-to-End): Commonly written as "E2E," not "e2e."
        PHPCS: Stays "PHPCS," not "phpcs."
        PHPStan: Stays "PHPStan," not "phpstan."
        YAML, JSON: These file format acronyms are usually uppercase. It's common to see "YAML" and "JSON" remain uppercase.
        WP-CLI: This should remain "WP-CLI," not "wp-cli."
        NPM: Typically written as "npm" in lowercase by the community, but if originally uppercase, it might need to remain consistent. The original usage in the documentation was "NPM," which is often stylized as "npm." If the original text used "NPM," then consistency with the original would be ideal, or consider using the official stylization "npm."
        CI/CD: These are standard acronyms. They should remain "CI/CD," not "ci/cd."

    Special Terms and Proper Nouns
        SemGrep: Brand name of a tool, usually capitalized as "Semgrep" or "SemGrep" depending on their official styling. In the original text, it appeared as "SemGrep." It should remain in that exact form.
        PHPCS: As above, always uppercase.
```

Then I had to do some back-and-forth with it to keep acronyms capitalized (anything that was originally two uppercase letters) and brands (WooCommerce, etc). After I (and it) was satisfied with the results I repeated the process for lists (such as `- **Foo**`, thus the two commits linked above.

File: docs/managed-tests/validation.md (hunk 4 above)
Comment (Luc45) [reply docs/managed-tests/validation.md]: Fixed, check screenshot below

File: docs/managed-tests/plugin-check.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# wordpress.org Plugin Check tests
+
+Tthe [wordpress.org automated plugin check](https://wordpress.org/plugins/plugin-check) tool against your extension. This optional test focuses on guidelines and patterns important for submitting plugins to the WordPress.org directory.
```
Comment (Luc45) [inline docs/managed-tests/plugin-check.md]:
```suggestion
The [wordpress.org automated plugin check](https://wordpress.org/plugins/plugin-check) tool against your extension. This optional test focuses on guidelines and patterns important for submitting plugins to the WordPress.org directory.
```

**Comment by zhongruige** [comment]
@Luc45 we mention for the custom tests and local env:

> The custom E2E tests feature is available as early access.

I'm thinking we can probably remove this wording now, what do you think?

File: docs/managed-tests/plugin-check.md (hunk 8 above)
Comment (zhongruige) [reply docs/managed-tests/plugin-check.md]: Ack! Good catch.

**Comment by Luc45** [comment]
I think so, too. On it.

**Review by zhongruige — APPROVED** [review]
Just finished going over this again and I believe it's looking good to merge and deploy. We can continue to tweak as needed, but so far seems the links are working as expected and made and pushed in a few adjustments, so i think we're in a good spot.

**Review by singerb — APPROVED** [review]
Yep, as mentioned yesterday I think the overall structure is much better, and with those tweaks the managed tests docs are looking good again too. I pushed some minor fixes to those pages but otherwise I think it's looking good too.

**Pull request merged**